### DISCUSSION TITLE
graph looks like it's an S rank actually it's an A rank.

### DISCUSSION THREAD

**Discussion by Inconsist** [ppy/osu#22501] [Q&A]
Image speaks for itself. Am I just picky or is this a problem? Also I feel like ive seen something "like it" before but i couldnt find the post so sorry if this is a repost.

![image](https://user-images.githubusercontent.com/87947227/216456078-6c8a2bf3-06f2-45d4-8816-5de171462da4.png)

**Comment by peppy** [comment]
Offsetting the "separator" lines to the start of the next zone may read better to the user?

**Reply by monochrome22** [reply]
That would change the problem from "94.99% looks like it should be S rank" to "95.00% looks like it should be A rank".
The most clean solution would be to just have the ring jump over the separator lines, like this:


![accbar](https://user-images.githubusercontent.com/122969285/216765948-780f718e-9fd2-40d7-b086-0fc9a838120a.png)
(More clear illustration in a later comment)

**Reply by bdach** [reply]
I dunno, it may work for this one particular case shown in the OP, but what about SS (the hot pink part)? It's really a single infinitesimal point on this inner ring rather than an actual area as shown.

**Reply by monochrome22** [reply]
I'm not sure what you mean. This is what 99.98% currently looks like: 

![9998](https://user-images.githubusercontent.com/122969285/216769378-796b8b12-cac8-4923-8112-ad7636fb69fe.png)

For SS, it just connects the ring. Wouldn't it work just fine?

**Reply by bdach** [reply]
That's inside the pink zone no? So it's an S-rank that looks like an SS? I imagine 99.99% would look even worse.

**Reply by monochrome22** [reply]
Yes, right now it looks like it is. It's actually not inside the pink zone, it just looks like it is because the ring is on the separator. 
That's the problem we're trying to solve. This is what looks like zoomed in: 

![zoomed](https://user-images.githubusercontent.com/122969285/216770021-2be43db7-ffa2-4c83-a7ab-dd6de08b12d3.png)

With my solution it wouldn't be on the separator, 99.99% would be at the end of the blue line, and for SS it would just connect the ring. In case it's not clear, the screenshots I'm showing are from how the game currently is, not how it would look like with my solution. 

Both mine and peppy's solution would look the same at 99.99% and SS, but mine would work better overall because the ring would never be on a separator, though I would imagine it would be more tedious to implement. It's up to developers what they choose.

**Reply by bdach** [reply]
My point is that mathematically speaking, there should be no pink area at all. It would just be a 1px line at most if we want to be accurate. Is that what you're suggesting?

**Reply by monochrome22** [reply]
That's not what I'm suggesting. This discussion is only about making the ring point to a more accurate location on the graph. Redesigning the graph would be a separate topic. 

I hope this explains it: 
![explanation](https://user-images.githubusercontent.com/122969285/216812309-2ada73db-c22c-4151-a385-4300edba3abe.png)

**Reply by peppy** [reply]
One of the proposed solutions is a lot harder to implement. If you're willing to give your approach a go and can implement in a simple way then that sounds fine. Else I'd go for the easier option.

**Reply by monochrome22** [reply]
I'm not a programmer, so I can't implement anything.
Though I had an idea on how to implement it. The separator is 0.2% accuracy wide, so you could draw the ring as if accuracy was 94.80 if it's more than 94.80 or less than 95.00. Apply the same to other grades. It would mean 94.81 and 94.99 look the same, but that would be less noticeable than having the ring on a separator. 
I'm not familiar with code, so if this is obviously not viable, then I'll avoid making further suggestions on how to implement something code-wise.

**Comment by Walavouchey** [comment]
with a few lines of additional logic so the gauge doesn't land in the gaps, i've managed the following. it effectively does what monochrome22 described [above](https://github.com/ppy/osu/discussions/22501#discussioncomment-4880036).

| 89.9% B | 90% A |
| :-: | :-: |
| ![image](https://user-images.githubusercontent.com/36758269/217219435-96556b41-7a44-42dd-a89d-13f4b6f1f71f.png) | ![image](https://user-images.githubusercontent.com/36758269/217219765-649eb991-2daf-4b6d-97ee-ad1ce9ac1c1e.png) |

demo of all cases at the edges of the sectors:

https://user-images.githubusercontent.com/36758269/217234101-fa3af1a4-a023-4aa2-8ffc-e5b1b30ffe84.mp4

branch at https://github.com/ppy/osu/compare/master...Walavouchey:osu:ranking-accuracycircle-readability. @peppy if this looks like a good direction i can make a pr

---

as an extra, i think the positions of the rank badges should be adjusted to perhaps more intuitive positions, i.e. the centre of their corresponding sectors.

Before:

https://user-images.githubusercontent.com/36758269/217233067-35e0cc47-898b-479a-a87e-84080c8d286d.mp4

After:

https://user-images.githubusercontent.com/36758269/217233345-09966852-613b-41ca-88ca-6e979bda76a3.mp4

**Reply by monochrome22** [reply]
The .99's look perfect, but the .00's look a bit too far up, would it look better if they were lowered slightly? Just 1-2 pixels.

Also the adjusted C and B badges look much better, but A S and SS might be better on old, because of SS being straight at the top. Maybe only slightly moving A and S toward the middle would be best, without dislocating the SS.

**Reply by Walavouchey** [reply]
> The .99's look perfect, but the .00's look a bit too far up, would it look better if they were lowered slightly? Just 1-2 pixels.

| Before | After |
| :-: | :-: |
| ![image](https://user-images.githubusercontent.com/36758269/217263147-89f1d165-eac4-4bdd-8f03-2e328e9e0629.png) | ![image](https://user-images.githubusercontent.com/36758269/217262377-4be57714-d7bc-4915-9c4b-fd8dd965f099.png) |

> Maybe only slightly moving A and S toward the middle would be best, without dislocating the SS.

| Before | After |
| :-: | :-: |
| ![image](https://user-images.githubusercontent.com/36758269/217265361-3ff881d1-e933-4089-a327-20d8419489cd.png) | ![image](https://user-images.githubusercontent.com/36758269/217264547-987d394c-8dab-44f1-86c2-d69307e0b7d0.png) |

yeah seems better

**Reply by peppy** [reply]
Visually that could work. I was thinking more that the actual progress (the animation) would "skip" over the holes, but that could be even harder to implement.

That said, looking at your code it's very hard to follow the math being done. I'd see if you can tidy it up by splitting out methods or variables further.. else we'll figure something out during review. Should be fine to go in this direction though.
